src/map/utils/charutils.cpp

```diff
@@ -860,13 +860,13 @@ namespace charutils
         charutils::LoadInventory(PChar);
 
         CalculateStats(PChar);
+        blueutils::LoadSetSpells(PChar);
         BuildingCharSkillsTable(PChar);
         BuildingCharAbilityTable(PChar);
         BuildingCharTraitsTable(PChar);
         jobpointutils::RefreshGiftMods(PChar);
 
         // Order matters as these use merits and JP gifts
-        blueutils::LoadSetSpells(PChar);
         puppetutils::LoadAutomaton(PChar);
 
```

Comment:
Ok, so, after speaking with Jmcmorris (and reading his PR)...

> This also fixes an issue with BLU JP gift JP_BLUE_MAGIC_POINT_BONUS is not applied properly when loading a character.

As you can see, simply reverting this change would re-introduce the issue.

After looking at other places in the code, it seems that `jobpointutils::RefreshGiftMods(PChar);` could be moved up, right below `CalculateStats(PChar);`

So the final order would be the following...
```c++
        CalculateStats(PChar);
        jobpointutils::RefreshGiftMods(PChar);
        
        // This must come after refreshing gift modifiers, but before loading job traits.
        blueutils::LoadSetSpells(PChar);

        BuildingCharSkillsTable(PChar);
        BuildingCharAbilityTable(PChar);
        BuildingCharTraitsTable(PChar);

        // Order matters as these use merits and JP gifts
        puppetutils::LoadAutomaton(PChar);
```

Of course, this would require testing.

Reply:
I did more testing and found a different issue altogether, as of commit c944d914bcdfafcf9114143e825ebf757a280db9, with my patch applied and with the order you posted here.

I found that I don't get the extra Blue Magic point immediately after spending JP, I have to zone or log out to get them.

Reply:
If this change doesn't introduce that issue, or rather, if that issue already exists, then I would say it's "out of scope" of this PR.

Reply:
> I found that I don't get the extra Blue Magic point immediately after spending JP, I have to zone or log out to get them.

Sounds like we fail to send something to the client in some other part of the core